src/shogun/machine/Machine.cpp

```diff
@@ -310,3 +306,23 @@ CLatentLabels* CMachine::apply_locked_latent(SGVector<index_t> indices)
 			"for %s\n", get_name());
 	return NULL;
 }
+
+uint64_t CMachine::supports() const 
+{
+	return m_supports;
+}
+
+bool CMachine::supports(uint64_t property_mask)
+{
+	return (m_supports & property_mask) == property_mask;
+}
+
+void CMachine::add_support(uint64_t property_mask)
+{
+	m_supports |= property_mask;
+}
+
+void CMachine::remove_support(uint64_t property_mask)
```

Comment:
Not sure we need this one?

Reply:
also I think the operation should be like here https://github.com/shogun-toolbox/shogun/blob/6434fe2df14bb0dbc34c27b2da4c7147f67b1159/src/shogun/base/AnyParameter.h#L136
You should use the complement

Reply:
in case a child class doesn't support a property of its parent class